hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackPlanActionExecutor.java

```diff
@@ -129,9 +138,36 @@ protected Option<HoodieRollbackPlan> requestRollback(String startRollbackTime) {
     }
   }
 
+  private boolean canProceedWithRollback(HoodieInstant rollbackInstant) {
+    if (config.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()) {
+      // check for concurrent rollbacks. i.e if the commit being rolledback is already rolled back, we can bail out.
+      HoodieTableMetaClient reloadedMetaClient = HoodieTableMetaClient.reload(table.getMetaClient());
```

Comment:
do we need to rebuild the meta client?

Reply:
we could just reload the active timeline and leverage that.

Reply:
actually the reason is 2 call stacks down. 
in L 152 we call 
validateForLatestTimestampWithoutReload() w/ reloaded meta client as argument. 

eventually this calls into 

TimestampUtils.validateForLatestTimestamp

within this method, we might need to reload the timeline in most cases. Only in case of this rollback flow, we do not need to reload the timeline. 
but to reload the timeline, we need an instant of "HoodieTableMetaClient" and so we could do metaClient.reloadActiveTimeline(). 

In other words, we can't just take in an activeTimeline as an argument to TimestampUtils.validateForLatestTimestamp

I did think about adding two Optional arguments to this method. but felt it may not be elegant.

Reply:
I have attempted a fix to make this elegant. you can check it out